### PR TITLE
Fix defects reported by coverity.

### PR DESCRIPTION
**Pull request by bah07**
|Related issue|
|---|
|#5910|

## Description

This PR fixes the defects found by coverity described in issue #5910 

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Review logs syntax and correct language

### CODE REVIEW

File: src/syscheckd/seechanges.c [hunk 1]
```diff
@@ -569,7 +569,12 @@ int seechanges_estimate_compression(const float file_size) {
 }
 
 void seechanges_modify_estimation_percentage(const float compressed_size, const float uncompressed_size) {
-    float compression_rate = 1 - (compressed_size / uncompressed_size);
+    float compression_rate;
+
+    if (uncompressed_size <= 0) {
+        return;
+    }
+    compression_rate = 1 - (compressed_size / uncompressed_size);
```
Comment (vikman90) [inline src/syscheckd/seechanges.c:577]: If `FileSize()` fails (`-1`) for the first parameter, `compression_rate` will be negative and the condition below will always match.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -845,38 +845,40 @@ void wm_vuldet_free_cve_node(void *data){
...
-    os_free(report->operation_value);
-    os_free(report->condition);
-    os_free(report->arch);
-    // Agent data
-    os_free(report->agent_id);
-    os_free(report->agent_name);
-    os_free(report->agent_ip);
-    // Metadata
-    os_free(report);
+    if (report) {
```
Comment (vikman90) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:848]: I think this change is clearer:
```c
if (report == NULL) {
    return;
}
```

**Review by vikman90 — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 2 above)
Comment (bah07) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: Done.

**Review by vikman90 — APPROVED** [review]
LGTM!

**Pull request merged**